ui/pages/confirmations/components/alerts/alert-modal/alert-modal.tsx

```diff
@@ -0,0 +1,199 @@
+    label: string;
+    /** The function to be executed when the custom button is clicked. */
+    onClick: () => void;
+    /** The variant of the custom button. */
+    variant?: string;
+  };
+};
+
+function getSeverityStyle(severity: Severity) {
+  switch (severity) {
+    case Severity.Warning:
+      return {
+        background: BackgroundColor.warningMuted,
+        icon: IconColor.warningDefault,
+      };
+    case Severity.Danger:
+      return {
+        background: BackgroundColor.errorMuted,
+        icon: IconColor.errorDefault,
+      };
+    default:
+      return {
+        background: BackgroundColor.infoMuted,
+        icon: IconColor.infoDefault,
+      };
+  }
+}
+
+export function AlertModal({
+  ownerId,
+  handleButtonClick,
+  alertKey,
+  onClose,
+  customButton,
+}: AlertModalProps) {
+  const t = useI18nContext();
+  const handleClose = useCallback(() => {
+    onClose();
+  }, [onClose]);
+  const { alerts, isAlertConfirmed, setAlertConfirmed } = useAlerts(ownerId);
+
+  const selectedAlert = alerts.find((alert) => alert.key === alertKey);
+
+  if (!selectedAlert) {
+    return null;
+  }
+  const isConfirmed = isAlertConfirmed(selectedAlert.key);
+  const severityStyle = getSeverityStyle(selectedAlert.severity);
+
+  return (
+    <Modal isOpen onClose={handleClose}>
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader onClose={handleClose}>
+          <Box
+            gap={3}
+            display={Display.Block}
+            alignItems={AlignItems.center}
+            textAlign={TextAlign.Center}
+          >
+            <Icon
+              name={
+                selectedAlert.severity === Severity.Info
+                  ? IconName.Info
+                  : IconName.Danger
+              }
+              size={IconSize.Xl}
+              color={severityStyle.icon}
+            />
+            <Text
+              variant={TextVariant.headingSm}
+              color={TextColor.inherit}
+              marginTop={3}
+            >
+              {selectedAlert.reason || t('alerts')}
+            </Text>
+          </Box>
+        </ModalHeader>
+        <ModalBody>
+          <Box
+            key={selectedAlert.key}
+            display={Display.InlineBlock}
+            padding={2}
+            width={BlockSize.Full}
+            backgroundColor={severityStyle.background}
+            gap={2}
+            borderRadius={BorderRadius.SM}
+          >
+            <Text variant={TextVariant.bodySm}>{selectedAlert.message}</Text>
+            {selectedAlert.alertDetails &&
+            selectedAlert.alertDetails?.length > 0 ? (
+              <Text variant={TextVariant.bodySmBold} marginTop={1}>
+                {t('alertModalDetails')}
+              </Text>
+            ) : null}
+
+            <Box
+              as="ul"
+              className={'alert-modal__alert-details'}
+              paddingLeft={6}
+            >
+              {selectedAlert.alertDetails?.map((detail, index) => (
```

Comment:
As this is a big block of JSX, is it worth splitting it into a few separate functions for easier maintenance?

We can keep the basic modal structure here but maybe call into local functions like:

- `Header`
- `AlertDetails`
- `AcknowledgeCheckbox`
- `AcknowledgeButton`

Reply:
I loved this idea, done :1st_place_medal: